Non-fully initialised MonitoringHub throws exception from __del__ finalizer

When a MonitoringHub has not been fully initialised (in my case due to a missing parameter), the `_-del__` finalizer throws an error as it incorrectly assumes that it will be called against a fully initialised object.

This creates a spurious stack trace.

```
E   TypeError: __init__() missing 1 required positional argument: 'hub_address'

parsl/tests/configs/htex_local.py:25: TypeError
----------------------------------------- Captured stderr setup ------------------------------------------
Exception ignored in: <object repr() failed>
Traceback (most recent call last):
  File "/home/benc/parsl/src/parsl/parsl/monitoring/monitoring.py", line 250, in __del__
    self.close()
  File "/home/benc/parsl/src/parsl/parsl/monitoring/monitoring.py", line 234, in close
    if self.logger:
AttributeError: 'MonitoringHub' object has no attribute 'logger'
```

--- a/parsl/monitoring/db_manager.py
+++ b/parsl/monitoring/db_manager.py
@@ -190,9 +190,6 @@ class Database(object):
             PrimaryKeyConstraint('task_id', 'run_id', 'timestamp'),
         )
 
-    def __del__(self):
-        self.session.close()
-
 
 class DatabaseManager(object):
     def __init__(self,

--- a/parsl/monitoring/monitoring.py
+++ b/parsl/monitoring/monitoring.py
@@ -121,9 +121,6 @@ class UDPRadio(object):
             return False
         return x
 
-    def __del__(self):
-        self.sock.close()
-
 
 class MonitoringHub(RepresentationMixin):
     def __init__(self,
@@ -277,9 +274,6 @@ class MonitoringHub(RepresentationMixin):
             self.logger.info("Terminating Hub")
             self.queue_proc.terminate()
             self.priority_msgs.put(("STOP", 0))
-
-    def __del__(self):
-        self.close()
 
     @staticmethod
     def monitor_wrapper(f, task_id, monitoring_hub_url, run_id, sleep_dur):

